client/src/components/QueryComponent/QueryComponent.js

```diff
@@ -62,7 +76,30 @@ class QueryComponent extends Component {
   }
 
   onChange = evt => {
+    let count = 0;
     this.setState({ selectedQuery: evt.target.value, selectedQueryKey: null });
+    evt.target.value.split("").map(character => {
+      if (character === "\n") {
+        count++;
+      }
+      return count;
+    });
+    this.setState({ rowCount: count + 1 });
+  }
+
+  onFocusGained = () => {
+    if (this.state.multilineHolder) {
+      this.setState({ multiline: true });
+    }
+  }
+
+  onFocusLost = () => {
+    this.setState({ disabled: true });
+    this.setState({ multiline: false });
```

Comment:
Is there any reason we call setState twice? I.e. rather than setting both properties in a single call?

Reply:
Change Ready